app/models/badge.rb

```diff
@@ -1,22 +1,28 @@
 class Badge < ApplicationRecord
   include CanBeHooked
-  has_paper_trail versions: { class_name: "AuditLog" }
+  has_paper_trail versions: { class_name: 'AuditLog' }
   belongs_to :badge_type
   belongs_to :team, optional: true
   has_many :badge_assignments, dependent: :destroy
   has_many :people, through: :badge_assignments
+  enum :permission, { added_by_admin: 0, added_by_manager: 1, added_by_current_member: 2, has_account: 3 }
```

Comment:
Not sure if this idea complicates things, but maybe have a badge that can be "self add"-able? It might be useful for like sign-ups for notifications or something..